pkg/controller/integration/monitor.go

```diff
@@ -407,9 +407,11 @@ func getIntegrationDigest(envs []corev1.EnvVar) string {
 	return ""
 }
 
-func findIntegrationContainer(spec corev1.PodSpec) *corev1.Container {
+// findIntegrationContainer find if present the integration container in the pod spec using the integration specifications.
+func findIntegrationContainer(environment *trait.Environment, spec corev1.PodSpec) *corev1.Container {
+	integrationContainerName := environment.GetIntegrationContainerName()
```

Comment:
Any reason why not to expect the integration container name here and passing the whole `environment` along all the methods? I'd be more cautious just passing what we need and keep more readable the func signature.

Reply:
You are right, I will move the `integrationContainerName := environment.GetIntegrationContainerName()` code up to the `filterPodsByReadyStatus` method.